tests/serializable.cpp

```
@@ -0,0 +1,95 @@
+TEST(serializable, node_adapter) {
+  auto param = std::string{};
+
+  auto yaml_node = YAML::Node{};
+  auto yaml_adapter = details::NodeAdapter<YAML::Node>{yaml_node};
+
+  auto ret2 = yaml_adapter.get_param("", param);
+  ASSERT_FALSE(ret2.has_value()) << "Expected error for missing key on empty node";
+  ASSERT_TRUE(param.empty());
+}
+
+TEST(serializable, node_adapter_type_mismatch) {
```

Clang (14.0.6)

[warning] all parameters should be named in a function

(readability-named-parameter)